cgsmiles/drawing_utils.py

```diff
@@ -0,0 +1,193 @@
+from collections import defaultdict
+import math
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.colors import ListedColormap
+
+def angle_between(v1, v2):
+    cos_theta = np.dot(v1, v2) / (np.linalg.norm(v1) * np.linalg.norm(v2))
+    angle_rad = np.arccos(np.clip(cos_theta, -1.0, 1.0))
+    return np.degrees(angle_rad)
```

Comment:
Why do you need the clip? Rounding issues? Would that get better if you normalize the vectors before the dot product?